brian2/spatialneuron/spatialneuron.py

```diff
@@ -600,24 +600,12 @@ def __init__(self, group, method, clock, order=0):
         self._morph_idxchild = group.flat_morphology.morph_idxchild
         self._starts = group.flat_morphology.starts
         self._ends = group.flat_morphology.ends
-        self._prepare_codeobj = None
 
     def before_run(self, run_namespace):
-        # execute code to initalize the data structures
-        if self._prepare_codeobj is None:
-            self._prepare_codeobj = create_runner_codeobj(self.group,
-                                                          '', # no code,
-                                                          'spatialneuron_prepare',
-                                                          name=self.name+'_spatialneuron_prepare',
-                                                          check_units=False,
-                                                          additional_variables=self.variables,
-                                                          run_namespace=run_namespace)
-        self._prepare_codeobj()
+        super(SpatialStateUpdater, self).before_run(run_namespace)
         # Raise a warning if the slow pure numpy version is used
-        # For simplicity, we check which CodeObject class the _prepare_codeobj
-        # is using, this will be the same as the main state updater
         from brian2.codegen.runtime.numpy_rt.numpy_rt import NumpyCodeObject
-        if isinstance(self._prepare_codeobj, NumpyCodeObject):
+        if isinstance(self.code_objects[0], NumpyCodeObject):
```

Comment:
This ``self.code_objects[0]`` makes me slightly nervous. Can the different items in the ``code_objects`` list have different types or are they guaranteed to all have the same type? If the former, then hard coding the 0 to refer to some particular thing feels like it might break later. If the latter, it's OK but perhaps we should think of adding some assertions for that?

Reply:
In general there's no guarantee. This could potentially be an issue in situations where we allow the user to specify a `codeobj_class` manually (which we do for `NeuronGroup` but not in `SpatialNeuron` – not sure we had a reason for that). Here, we only have a single object in the list, so there's no potential for confusion.